spec/expressions.md

```diff
@@ -663,48 +710,52 @@ F(10, 20, new object\[\] {});\
+When a generic method is called without specifying type arguments, a ***type inference*** process attempts to infer type arguments for the call. The presence of type inference allows a more convenient syntax to be used for calling a generic method, and allows the programmer to avoid specifying redundant type information. Given the method declaration:
 
 class Chooser\
 {\
 static Random rand = new Random();
 
-public static T Choose&lt;T&gt;(T first, T second) {\
+public static T Choose<T>(T first, T second) {\
 return (rand.Next(2) == 0)? first: second;\
 }\
 }
 
 it is possible to invoke the Choose method without explicitly specifying a type argument:
 
-int i = Chooser.Choose(5, 213); // Calls Choose&lt;int&gt;
+int i = Chooser.Choose(5, 213); // Calls Choose<int>
 
-string s = Chooser.Choose("foo", "bar"); // Calls Choose&lt;string&gt;
+string s = Chooser.Choose("foo", "bar"); // Calls Choose<string>
 
-Through type inference, the type arguments int and string are determined from the arguments to the method. *end example*\]
+Through type inference, the type arguments int and string are determined from the arguments to the method. 
 
 Type inference occurs as part of the binding-time processing of a method invocation (§12.7.6.2) and takes place before the overload resolution step of the invocation. When a particular method group is specified in a method invocation, and no type arguments are specified as part of the method invocation, type inference is applied to each generic method in the method group. If type inference succeeds, then the inferred type arguments are used to determine the types of arguments for subsequent overload resolution. If overload resolution chooses a generic method as the one to invoke, then the inferred type arguments are used as the type arguments for the invocation. If type inference for a particular method fails, that method does not participate in overload resolution. The failure of type inference, in and of itself, does not cause a binding-time error. However, it often leads to a binding-time error when overload resolution then fails to find any applicable methods.
 
 If each supplied argument does not correspond to exactly one parameter in the method (§12.6.2.2), or there is a non-optional parameter with no corresponding argument, then inference immediately fails. Otherwise, assume that the generic method has the following signature:
 
-T~r~ M&lt;X~1~…X~n~&gt;(T~1~ p~1~ … T~m~ p~m~)
+T~r~ M<X~1~…X~n~>(T~1~ p~1~ … T~m~ p~m~)
 
-With a method call of the form M(E~1~ …E~m~) the task of type inference is to find unique type arguments S~1~…S~n~ for each of the type parameters X~1~…X~n~ so that the call M&lt;S~1~…S~n~&gt;(E~1~…E~m~)becomes valid.
+With a method call of the form M(E~1~ …E~m~) the task of type inference is to find unique type arguments S~1~…S~n~ for each of the type parameters X~1~…X~n~ so that the call M<S~1~…S~n~>(E~1~…E~m~)becomes valid.
```

Comment:
Similar to above:

```suggestion
With a method call of the form `M(E1...Em)` the task of type inference is to find unique type arguments `S1...Sn` for each of the type parameters `X1...Xn` so that the call `M<S1...Sn>(E1...Em)` becomes valid.
```